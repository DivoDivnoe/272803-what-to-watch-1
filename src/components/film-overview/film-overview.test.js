import React from 'react';
import {create} from 'react-test-renderer';
import FilmOverview from './film-overview.jsx';

const mock = {
  film: {
    name: `Major Payne`,
    description: `very interesting film`,
    director: `Steven Spielberg`,
    starring: [`Andrey Ivanov`, `Sergey Rubets`],
    rating: 10,
    scoresCount: 100000,
    backgroundImage: ``,
    backgroundColor: `cyan`,
    released: 2019,
    posterImage: ``,
    genre: `Comedy`,
    previewVideoLink: ``,
    previewImage: ``,
  },
};

describe(`FilmOverview component`, () => {
  it(`renders correctly`, () => {
    const {film} = mock;

    const tree = create(
        <FilmOverview film={film} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
