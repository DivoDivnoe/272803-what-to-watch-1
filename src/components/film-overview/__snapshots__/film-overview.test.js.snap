// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmOverview component renders correctly 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      10
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Awesome
      </span>
      <span
        className="movie-rating__count"
      >
        100000
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      very interesting film
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Steven Spielberg
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Andrey Ivanov, Sergey Rubets
         and other
      </strong>
    </p>
  </div>,
]
`;
